Source: https://github.com/systemd/systemd/commit/6d1e2ddd59f929ce98e400cb14fba7a873e2b409
Status: merged, fixed in v232
Reason: fix test

# ----------

From 6d1e2ddd59f929ce98e400cb14fba7a873e2b409 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppymaster@gmail.com>
Date: Sun, 31 Jul 2016 21:50:50 -0400
Subject: [PATCH] test-path-util: check for /lt-test-path-util or
 /test-path-util (#3841)

Depending on how binutils was configured and the --enable-fast-install
configure option, the test binary might be called either name.

Fixes: https://github.com/systemd/systemd/issues/3838
---
 src/test/test-path-util.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/test/test-path-util.c b/src/test/test-path-util.c
index 6094d4c..164a10d 100644
--- a/src/test/test-path-util.c
+++ b/src/test/test-path-util.c
@@ -114,7 +114,8 @@ static void test_find_binary(const char *self) {
 
         assert_se(find_binary(self, &p) == 0);
         puts(p);
-        assert_se(endswith(p, "/lt-test-path-util"));
+        /* libtool might prefix the binary name with "lt-" */
+        assert_se(endswith(p, "/lt-test-path-util") || endswith(p, "/test-path-util"));
         assert_se(path_is_absolute(p));
         free(p);
 
